packages/java-edition/src/mcfunction/tree/argument.ts

```diff
@@ -150,16 +150,46 @@ export interface MinecraftMobEffectArgumentTreeNode
 {
 	parser: 'minecraft:mob_effect'
 }
+export interface NbtParserProperties extends Record<string, unknown> {
+	/**
+	 * The NBT checker should check this argument by dispatching on `dispatcher`
+	 * with a static index inferred from the argument that is at the index
+	 * `dispatchedBy` relative to the current argument and optionally indexing
+	 * on the dispatched type by indices inferred from the argument that is at the
+	 * index `indexedBy` relative to the current argument.
+	 */
+	dispatcher:
+		| 'minecraft:block_entity'
+		| 'minecraft:entity'
+		| 'minecraft:storage'
+	/**
+	 * Could point to a vec3, an entity, an entity resource location, or a storage
+	 * resource location.
+	 *
+	 * @see {@link NbtParserProperties.dispatcher}
+	 */
+	dispatchedBy: number
```

Comment:
Is there a way we can reference the argument name here instead of a (relative?) index? The magic `-4` values feel dangerous to me.